py/embedded-server/deephaven_server/start_jvm.py

```diff
@@ -98,5 +100,6 @@ def start_jvm(
         jvm_options=jvm_args,
         # config_file=config_file,
         config=config)
+    deephaven_internal.server_ready()
```

Comment:
this is too early - we should run this only after the Server's constructor has completed (so we know the logging, config is all finished, not just able to start)

Reply:
It might even be worth not doing this at all, and instead letting the java JpyInit set this, since it knows everything is done on the java side.

Reply:
The more I think about it, the more I like that - we can be certain that way that everything ran and was happy on the java side, rather than just assuming it will work out, plus, two fewer lines of code to worry about here.

There is no downside from the perspective of helping a user understand what is wrong - the check will still fail if they try to import deephaven and this flag hasn't been set. If the flag failed to be set (from importing deephaven_server.Server or from instantiating a Server), then the user has that log to look at.

Reply:
I think that's a good idea, but unfortunately that will require a bit more work (I'll forge ahead to see how much actual work...).

During startup we do:

```java
public PythonDeephavenSession( ... ) {
...
this.module = (PythonScriptSessionModule) PyModule.importModule("deephaven.server.script_session")
                .createProxy(CallableKind.FUNCTION, PythonScriptSessionModule.class);
 ```
 
 And this causes `deephaven` module to be imported.

As the code exists right now in the PR, variable state I'm really setting is "was/has Deephaven initialized the JVM & Python?". It may be worth having hooks both for that, and for "is the server ready" - and this might make the sphinx and test parts a little more explicit.

Reply:
Looks like even if `module` is made to be lazy, it's still called as part of `AbstractScriptSession.publishInitial`

```
at org.jpy.PyLib.importModule(Native Method)
2022-07-20T15:09:04.120006480Z 	at org.jpy.PyModule.importModule(PyModule.java:94)
2022-07-20T15:09:04.120008334Z 	at io.deephaven.integrations.python.PythonDeephavenSession.module(PythonDeephavenSession.java:336)
2022-07-20T15:09:04.120018182Z 	at io.deephaven.integrations.python.PythonDeephavenSession.createDiff(PythonDeephavenSession.java:228)
2022-07-20T15:09:04.120042338Z 	at io.deephaven.integrations.python.PythonDeephavenSession.createDiff(PythonDeephavenSession.java:56)
2022-07-20T15:09:04.120062015Z 	at io.deephaven.engine.util.AbstractScriptSession.applyDiff(AbstractScriptSession.java:139)
2022-07-20T15:09:04.120072144Z 	at io.deephaven.engine.util.AbstractScriptSession.publishInitial(AbstractScriptSession.java:107)
2022-07-20T15:09:04.120093665Z 	at io.deephaven.integrations.python.PythonDeephavenSession.<init>(PythonDeephavenSession.java:97)
2022-07-20T15:09:04.120107261Z 	at io.deephaven.server.console.python.PythonConsoleSessionModule.bindPythonSession(PythonConsoleSessionModule.java:31)
```

Reply:
While I think you are correct that actually checking that the "server is running" is "best", I think given the lifecycle of how things are initialized, that will be very tricky. I've renamed the state checking and added docs to make clear this is JVM tracking, and where it fits into the lifecycle of java-first and python-first processes.

I'm hoping it's "good enough" for now